src/main/java/com/strandls/observation/dao/ObservationDAO.java

```diff
@@ -178,10 +178,10 @@ public Long findTotalObservation() {
 	}
 
 	@SuppressWarnings({ "unchecked", "rawtypes" })
-	public List<Observation> getObservationCountForDatatable(Long datatableId) {
+	public Long getObservationCountForDatatable(String datatableId) {
 		Session session = sessionFactory.openSession();
 		List<Observation> observationList = new ArrayList<Observation>();
-		String hql = "from Observation o where o.dataTableId = :datatable";
+		String hql = "from Observation o where o.dataTableId = :datatable and o.isDeleted = false";
```

Comment:
https://github.com/strandls/biodiv-observation/blob/71437cd2352521825fe24f6eb0b532dfa5bd2fe3/src/main/java/com/strandls/observation/dao/ObservationDAO.java#L60

Use this ti bring the count for the datatable. its completely unnecessary for first bring out a list of object and then count it